File: frontend/src/api/news.ts
import apiClient from "./client";

export interface NewsItem {
  title: string;
  link: string;
  pub_date: string;
  source: string;
  image_url: string | null;
  category: string;
  matched?: boolean;
}

export interface NewsResponse {
  items: NewsItem[];
}

export const getNews = async (): Promise<NewsResponse> => {
  const { data } = await apiClient.get("/news/");
  return data;
};

export interface NewsSummary {
  summary: string | null;
  mode: "ai" | "unavailable";
}

export const getNewsSummary = async (
  item: Pick<NewsItem, "title" | "link" | "source" | "category">
): Promise<NewsSummary> => {
  const { data } = await apiClient.post("/news/summary", {
    title: item.title,
    link: item.link,
    source: item.source,
    category: item.category,
  });
  return data;
};

File: backend/app/schemas/news.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class NewsItem(BaseModel):
    title: str
    link: str
    pub_date: str
    source: str
    image_url: Optional[str] = None
    category: str
    matched: bool = False  # 내가 구독한 서비스와 관련된 뉴스인지

class NewsResponse(BaseModel):
    items: List[NewsItem]


class NewsSummaryRequest(BaseModel):
    title: str
    link: str
    source: str = ""
    category: str = ""


class NewsSummaryResponse(BaseModel):
    summary: Optional[str] = None
    mode: str  # "ai" | "unavailable"
